### DISCUSSION TITLE
Bug ao salvar ou atualizar qualquer cadastro

### DISCUSSION THREAD

**Discussion by llucasrn** [RamonSilva20/mapos#1747] [General]
Fiz a instalação em um ubuntu server, ele rodo bem, porém na hora de salvar ou atualizar qualquer cadastro, sou redirecionado para a página atualizada, nesse momento ele duplica a url.

Ex. O caminho é meudominio.com/mapos/index.php/cliente

Quando clico em salvar ou atualizar fica:
Meudominio.com/mapos/meu domínio.com/mapos/index.php/cliente

E vai para um tela de erro de página não encontrada 404.

![Screenshot_20220404-202647.png](https://user-images.githubusercontent.com/35467331/161648708-773985ff-9800-48a6-90ee-f180f31496d9.png)

**Comment by Pr3d4dor** [comment]
@llucasrn Você configurou corretamente o domínio?

**Reply by llucasrn** [reply]
Sim, seguir a explicação na descrição do projeto

**Comment by Pr3d4dor** [comment]
@llucasrn Eu acho acho que tem alguma configuração específica dessa hospedagem causando problema. Já tentou testar em outra?

**Reply by llucasrn** [reply]
Atualmente instalei na máquina local com XAMPP e em uma vps ubuntu com apache, em ambos os casos teve o mesmo bug

**Reply by llucasrn** [reply]
Mas vou ver outras opções de hospedagem local além do XAMPP

**Reply by lmainformatica** [reply]
tambem estou com esse erro, consegui resolver amigo?

**Reply by llucasrn** [reply]
Ainda, mas tive alguns problemas com o servidor, vou refazer a instalação e documentar passo a passo  para tentar encontrar o ponto onde deu errado, mas mesmo dando esse erro ele salva no banco de dados.

**Reply by lmainformatica** [reply]
verdade, amigo percebi isso tambem
anota meu zap ai 67992537461
tambem vou dar uma estudada

Att...Lincoln Moslaves

**Reply by lmainformatica** [reply]
> @llucasrn Eu acho acho que tem alguma configuração específica dessa hospedagem causando problema. Já tentou testar em outra?

amigo estou com o mesmo problema uso a locaweb
e tambem por local host, mais percebi que usando a tecla de voltar ela aparece cadastrado com sucesso..

**Comment by Pr3d4dor** [comment]
@llucasrn @lmainformatica Acho que a URL está configurada incorretamente.

https://www.youtube.com/watch?v=8joUGLpzUtE
https://www.youtube.com/watch?v=C6O-Te6qzog

**Comment by julianoninoka** [comment]
vá na pasta : \application\config --> $config['base_url'] = 'AQUI VOCE COLOCAR A URL CORRETA';    
vc deve ter colocado errado a url.

**Reply by llucasrn** [reply]
> vá na pasta : \application\config --> $config['base_url'] = 'AQUI VOCE COLOCAR A URL CORRETA';    
> vc deve ter colocado errado a url.
> 

Obrigado amigo, mas o erro continuo, porém preparei um ambiente de teste com as mesmas configurações do da produção, fiz a instalação do zero e funcionou normalmente, fiz alguns testes adicionando e editando produtos, clientes, serviços e ordens de serviço, nesse momento funcionou normal, então fui restaura o backup do banco de dados do que estava em produção, aí o problema voltou acontece, não sei o motivo, mas voltou, então já que não usamos o acesso para os clientes, fiz o backup da base de dados e coloquei no ambiente de testes e na produção refiz a instalação do zero, passei o acesso ao pessoal que refez o cadastro das ordens de serviço que ainda estava em aberto, a loja está usando as duas bases de dados por hora, a que restaurei o backup é apenas para consulta das ordens antigas e caso chegue um cliente querendo aciona a garantia, ele refaz o cadastro na hora, no novo, com os mesmo dados que estava no bando de dodos antigo, foi uma "solução" que encontrei até o momento, estou achando que tenha sido algo na atualização pois eu estava usando uma versão do map-os mais antiga, e o problema com o link começou depois que atualizei.

**Reply by julianoaraujonk** [reply]
vc pode enviar sua base de dados que esta tentando restaurar?

**Reply by llucasrn** [reply]
> vc pode enviar sua base de dados que esta tentando restaurar? 

Desculpa, mas não posso, a base de dados pertence ao um cliente, o mesmo não autorizou o compartilhamento.

Obrigado pelas dicas e ajuda.

**Comment by Khorzus** [comment]
Bom dia!

Estou precisando de uma ajuda! Quando vou cadastrar um novo fornecedor, eu digito o CNPJ, o Mapos busca diretinho mas na hora de clicar em "Salvar" dá erro!!! :(

Alguém consegue me ajudar?

A minha versão do Mapos é a 4.37.0
![erro mapos](https://user-images.githubusercontent.com/107259568/173075206-3951b6aa-95b5-4dee-a992-ab8b9d5c7043.jpg)

**Reply by llucasrn** [reply]
> Bom dia!
> 
> Estou precisando de uma ajuda! Quando vou cadastrar um novo fornecedor, eu digito o CNPJ, o Mapos busca diretinho mas na hora de clicar em "Salvar" dá erro!!! :(
> 
> Alguém consegue me ajudar?
> 
> A minha versão do Mapos é a 4.37.0
> ![erro mapos](https://user-images.githubusercontent.com/107259568/173075206-3951b6aa-95b5-4dee-a992-ab8b9d5c7043.jpg)
> 

É só quando vai salvar cliente e fornecedor, ou com outras partes também?

**Reply by Khorzus** [reply]
Descobri sozinho o que está acontecendo..... eu atualizei para versão mais recente que é a 4.37.0. E nessa versão, como vc pode ver na imagem acima, foi inserido nesse formulário de Cadastro de Cliente, o campo SENHA, e é esse campo que está dando pau e não deixando salvar quando clica no botão "Salvar".

Então o que eu fiz..... fui no site (https://github.com/RamonSilva20/mapos/releases) e baixei e instalei uma versão anterior: v4.36,2 e agora está dando certo! Agora não dá mais erro ao cadastrar um cliente/fornecedor.  ;)

**Comment by metricasolucoes** [comment]
![image](https://github.com/RamonSilva20/mapos/assets/165976929/4f90e560-109d-49e8-8a09-46f5cdfc83ff)

**Reply by metricasolucoes** [reply]
erro ao tentar cadastrar uma despesa no financeiro. Como resolver?

**Reply by Pr3d4dor** [reply]
@metricasolucoes precisa reinstalar e configurar o .env corretamente

**Reply by Pr3d4dor** [reply]
@metricasolucoes também precisa limpar o cache do navegador por conta do arquivo funcoesGlobal.js (ctrl + shift + r)

**Comment by johnteckinformatica** [comment]
> erro ao tentar cadastrar uma despesa no financeiro. Como resolver?

O meu resolveu instalando a versão direto do Code - master
